### ISSUE TITLE
Allow mail being changed

### ISSUE THREAD

**Issue by debuglevel** [bigbluebutton/greenlight#5593]
GL3 does not allow a user to change her mail address anymore.
Whenever a customer wants to change her mail address (and well, this happens for various reasons), we have to tell "sorry, please create a new account. And no, your rooms/URLs are lost."

It would be nice therefore, to allow users to have their mail changed.

**Comment by JeanPluzo** [comment]
Hi,

innocent question: couldn't you change the e-mail address directly in the database?
I know you can change some fields directly in de database in GLv2, but I haven't installed/tested GLv3.
OTOH it would be nice if a user can change her/his e-mail address without an admin's help.

Regards,
J.

**Comment by Lukas-Rotermund** [comment]
> > Hi,
> > innocent question: couldn't you change the e-mail address directly in the database? I know you can change some fields directly in de database in GLv2, but I haven't installed/tested GLv3. OTOH it would be nice if a user can change her/his e-mail address without an admin's help.
> 
> Yes, changing via postgres interface works, but self-service is more convenient for users and admins.

Hello,

how do I enter the postgres interface in Greenlight V3?
Tryed it the way I did in Greenlight V2, but it didn't work.

**Comment by bwildenhain** [comment]
> > > Hi,
> > > innocent question: couldn't you change the e-mail address directly in the database? I know you can change some fields directly in de database in GLv2, but I haven't installed/tested GLv3. OTOH it would be nice if a user can change her/his e-mail address without an admin's help.
> > 
> > 
> > Yes, changing via postgres interface works, but self-service is more convenient for users and admins.
> 
> Hello,
> 
> how do I enter the postgres interface in Greenlight V3? Tryed it the way I did in Greenlight V2, but it didn't work.

sudo docker exec -it postgres /usr/local/bin/psql -U postgres greenlight-v3-production

**Comment by sto20km** [comment]
> sudo docker exec -it postgres /usr/local/bin/psql -U postgres greenlight-v3-production

Hi. I have the same problem but I'm not an expert on postgres in GLv3.
What is the list of exact commands to run step be step to change user's e-mail address directly in the database?

**Comment by CatSema** [comment]
Hello @sto20km!

### 🔧 Guide: How to Change a User’s Email in PostgreSQL (Docker)  

#### **📌 1. Connect to the PostgreSQL Container**  
Open a terminal and run the following command to access PostgreSQL inside the Docker container:  
```bash
sudo docker exec -it postgres /usr/local/bin/psql -U postgres greenlight-v3-production
```
Where:  
- `postgres` – the container name.  
- `psql -U postgres greenlight-v3-production` – connects to the `greenlight-v3-production` database as the `postgres` user.  

#### **📌 2. Find the User by Their Current Email**  
Run the following SQL query to find the user’s `id`:  
```sql
SELECT id, email FROM users WHERE email = 'old_email@example.com';
```
Example output:  
```
                  id                  |           email           
--------------------------------------+---------------------------
 31193b1a-bfee-400f-b180-086d04def4c9 | old_email@example.com
(1 row)
```
Take note of the `id`, as it will be needed for the update.  

#### **📌 3. Update the Email Address**  
Run the following update query:  
```sql
UPDATE users 
SET email = 'new_email@example.com' 
WHERE id = '31193b1a-bfee-400f-b180-086d04def4c9';
```
If the update is successful, you will see:  
```
UPDATE 1
```

#### **📌 4. Verify the Changes**  
Run the following query to confirm the email has been updated:  
```sql
SELECT * FROM users WHERE id = '31193b1a-bfee-400f-b180-086d04def4c9';
```
Ensure that the `email` column now contains the new email address.

#### **📌 5. (Optional) Reset Email Verification**  
If the system requires email verification, reset the verification status:  
```sql
UPDATE users 
SET verified = FALSE, verification_digest = NULL, verification_sent_at = NOW()
WHERE id = '31193b1a-bfee-400f-b180-086d04def4c9';
```
If you want to log the user out (forcing them to reauthenticate):  
```sql
UPDATE users 
SET session_token = NULL, session_expiry = NOW()
WHERE id = '31193b1a-bfee-400f-b180-086d04def4c9';
```

#### **📌 6. Exit PostgreSQL**  
After making changes, exit `psql` by typing:  
```sql
\q
```
Then close the terminal.

---

### ✅ **Result**  
The user's email has been successfully updated, and they can now use the new email in the system! 🎯

**Comment by sto20km** [comment]
> ### 🔧 Guide: How to Change a User’s Email in PostgreSQL (Docker)
>...
> ### ✅ **Result**
> 
> The user's email has been successfully updated, and they can now use the new email in the system! 🎯

Hello, @CatSema!
Thanks a lot, this is exactly what I need! It works!

**Comment by Mai-Lapyst** [comment]
> Not a simple feature, though. Changing the password-reset email address of an account should only be allowed after entering the current password again, and only take effect once the new email is verified. The old email should also be notified about the change.
> 
> For external users, changing email should not be allowed at all. The email is used as a fallback for account correlation.

While I agree that an feature to allow any user to change their email is indeed problematic and needs enough verification steps, wouln't it be better to allow to change emails for any user atleast for (super)admins in the meantime to prevent providers / admins with a better way than the dangerous way over the postgresql cli where an oversight could do much more damage?

I also would contribute this feature if it's something that would get accepted.

**Comment by EmmyGraugans** [comment]
I agree with @Mai-Lapyst – at least allow admins to change the Email-Address (i.e. login-name) of users in the regular admin-interface. I tried it with the "brute" approach of enabling write-access to the `updateUserFormEmail` and changing the value before submit, but it appears to not be read on the backend.
